### DISCUSSION TITLE
MStar "Titania" MIPS Boards

### DISCUSSION THREAD

**Discussion by ElectrodeYT** [linux-chenxing/linux-chenxing.org#12] [General]
These boards were used in at least a few Vestel (OEM) TVs.

I found a linux kernel & mpatch stuff on GitHub (was a GPL Package it seems): https://github.com/Benjamin-Dobell/kogan-tv-gpl

It seems there are many versions of the titania boards.

I only have a (SilverCrest brand) TV with a Titania4 board. Getting a shell is quite easy: holding down OK on the control while plugging the TV in causes it to execute a file as U-Boot commands, meaning you can change the boot-delay variable and get a U-Boot shell, and then you can change the bootargs to inclue init=/bin/sh. The U-Boot / M-Boot seems a lot more "normal" then other M-Star devices i have. The busybox init file was supposed to launch a shell, but the /etc/init.d/rcS file is bugged and doesnt launch the main file in the background, causing it to never stop, and the shell to never start, this can be changed very easily. In addition, some files are not stripped / not needed, meaning enough space can be saved to put a "init-intercepting" file that can allow a pivot-root to USB (via a switch-root file on the USB, as the busybox on the TV is very restricted).

The linux kernel on it is too old to be of much use (compiled in 2010), the U-Boot is just as old but does support enough features that its probably not a serious problem. The only problem U-Boot has is that "usb start" is bugged: the only way to properly access USB is via running the update file.

The board has 128MB of ram, and 64MB of NAND Flash.

I have dumped the NAND flash (A SHARP Update file literally had the commands to dump it commented out).

**Comment by fifteenhex** [comment]
I think the MIPS devices are similar enough (aside from the CPU) that the drivers that are running on mainline for the ARM based chips might just work. I've never really worked with MIPS though so I don't know how much effort it would be to get u-boot/linux working enough to do something.

**Reply by ElectrodeYT** [reply]
Buildroot rootfs's do work (at least somewhat) with that really old version of linux (2.6) thats running on that TV. Compiling that kernel though is proving a challenge - MStar isnt really known for easy-to-patch kernels, although it seems that everything is here to patch a newer version of the Kernel to work.

**Reply by fifteenhex** [reply]
If you have a use case that doesn't involve driving the screen I think it might be possible to get mainline running without too much work assuming the MIPS specific parts for MStar can be forward ported.

If you have some time it'd be really nice to get a few bits on the wiki about the hardware you have.
